tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/jdbc/JdbcTableManager.java

```diff
@@ -73,18 +79,77 @@ public TableInstance createImmutable(JdbcTableDefinition tableDefinition, TableH
     {
         TableName tableName = createImmutableTableName(tableHandle);
         LOGGER.debug("creating immutable table {}", tableName);
-        dropTableIgnoreError(tableName); // ultimately we do not want to drop table here - if data did not change
-        createTable(tableDefinition, tableName);
-        JdbcTableDataSource dataSource = tableDefinition.getDataSource();
-        insertData(tableName, dataSource);
+        // TODO: drop and recreate table if the underlying data has changed
+        if (!tableExists(tableName)) {
+            createTable(tableDefinition, tableName);
+            JdbcTableDataSource dataSource = tableDefinition.getDataSource();
+            insertData(tableName, dataSource);
+        }
         return new JdbcTableInstance(tableName, tableDefinition);
     }
 
+    private boolean tableExists(TableName tableName)
+    {
+        Connection connection = queryExecutor.getConnection();
```

Comment:
we don't want to close the connection afterwards, since the single connection is used for all communication with Teradata